src/js.rs

```diff
@@ -0,0 +1,568 @@
+#[cfg(test)]
+mod tests;
+
+use crate::{
+    ast::*,
+    error::GleamExpect,
+    fs::{OutputFile, Utf8Writer},
+    line_numbers::LineNumbers,
+    pretty::*,
+    project::{self, Analysed},
+    type_::{
+        ModuleValueConstructor, PatternConstructor, Type, TypeVar, ValueConstructor,
+        ValueConstructorVariant,
+    },
+    Result,
+};
+use itertools::Itertools;
+use std::sync::Arc;
+
+const INDENT: isize = 4;
```

Comment:
Can we use 2? That seems to be common in JS

Reply:
Might do this at the end. Rather a pain as the test.rs file is automatically indenting to 4 as I move stuff around